### PR TITLE
Updated release notes for 1.7.0

### PR DESCRIPTION
**Pull request by rpoet-jh**

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+## Release v1.7.0
+### Date: May 30, 2024
+
+Release Manager: Russ Poetker, JHU
+
+This release focused on adding a GitHub workflow that will complete the release of all PASS components. The pass-core metadata schema service was changed as a first step in supporting InvenioRDM integration. A few smaller bugs were also fixed.
```
Comment (bbranan) [inline docs/release-notes.md:6]: I'd suggest adding a sentence or two here about the documentation work that was completed during this release (doc review manual, style guide, first-round of review). Maybe also note the background work happening to transition to Terraform.

Comment (markpatton) [reply docs/release-notes.md:6]: Agreed. That makes sense.

**Review by markpatton — APPROVED** [review]
Looks good.

File: docs/release-notes.md (hunk 1 above)
Comment (rpoet-jh) [reply docs/release-notes.md]: Done

**Review by bbranan — APPROVED** [review]

**Pull request merged**